junit-platform-commons/src/main/java/org/junit/platform/commons/util/ReflectionUtils.java

```diff
@@ -840,6 +842,22 @@ private static void findNestedClasses(Class<?> clazz, Set<Class<?>> candidates)
 	}
 
 	/**
+	 * Method comparator based upon JUnit4 org.junit.internal.MethodSorter implementation.
+	 */
+	private static int defaultMethodSorter(Method m1, Method m2) {
+		int i1 = m1.getName().hashCode();
+		int i2 = m2.getName().hashCode();
+		if (i1 != i2) {
+			return i1 < i2 ? -1 : 1;
+		}
```

Comment:
Just replace `return i1 < i2 ? -1 : 1;` with `return Integer.compare(i1, i2);`?

Or/and drop the entire duplicate hash value handling?